### PR TITLE
added cart mgmt wishlit mgmt, pages

### PR DESCRIPTION
**Pull request by MonikaJhingan**
pls review it urgently.

### CODE REVIEW

File: src/context/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  useEffect(() => {
+    const data = async () => {
+      try {
+        const response = await axios({
+          method: "get",
+          url: "/api/user/cart",
+          headers: { authorization: auth.token },
+        });
+        setItemsAdded(response.data.cart);
+      } catch (err) {
```
Comment (ashishnnnnn) [inline src/context/cart-context.js:21]: Good that you have handled error also.

File: src/components/CartProduct/CartProduct.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  const { moveToWishlist } = useWishlist();
+
+  return (
+    <div>
+      {itemsAdded &&
+        itemsAdded.map((product) => {
+          const { image, name, price, _id, qty, discount } = product;
+          return (
+            <div className="cart-card" key={_id}>
+              <img src={image} className="cart-image" />
```
Comment (ashishnnnnn) [inline src/components/CartProduct/CartProduct.jsx:23]: These Img can have alt attribute also.

File: src/context/wishlist-context.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      }}
+    >
+      {children}
+    </WishlistContext.Provider>
+  );
+};
+
+const useWishlist = () => useContext(WishlistContext);
+
+export { useWishlist, WishlistProvider };
```
Comment (ashishnnnnn) [inline src/context/wishlist-context.js:107]: Good that for wish-list,cart you have their own context.

**Review by ashishnnnnn — APPROVED** [review]

File: src/components/WishlistProduct/WishlistProduct.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+export const WishlistProduct = () => {
+  const { wishlistItems, removeFromWishlist, moveToCart, errorMsg } =
+    useWishlist();
+  return (
+    <div className="wishlist-items">
+      {errorMsg && <h1>item is already in the cart</h1>}
+      {wishlistItems.map((item) => {
+        const { image, name, price } = item;
+        return (
+          <div class="card card-dismiss">
```
Comment (kavyakunder) [inline src/components/WishlistProduct/WishlistProduct.jsx]: replace class with className in react

**Review by kavyakunder — APPROVED** [review]
Just replace class with className in WishlistProduct file.
The code is easily understandable and the separation of dividing components made it easy for review.

File: src/components/CartProduct/CartProduct.jsx (hunk 2 above)
Comment (MonikaJhingan) [reply src/components/CartProduct/CartProduct.jsx:23]: Ok will do

File: src/components/WishlistProduct/WishlistProduct.jsx (hunk 4 above)
Comment (MonikaJhingan) [reply src/components/WishlistProduct/WishlistProduct.jsx]: Ok thanks a lot will do

**Pull request merged**